ios/RNCAsyncStorage.m

```diff
@@ -167,6 +219,19 @@ @implementation RNCAsyncStorage
   NSMutableDictionary<NSString *, NSString *> *_manifest;
 }
 
++ (BOOL)requiresMainQueueSetup {
+  return YES;
```

Comment:
Why do we require main queue? Isn't `NSFileManager` thread safe?

Reply:
Indeed, I looked into it and I think I can return `NO` here since we are not using any UIKit elements. React-native tends to complain if you `init` without implementing `requiresMainQueueSetup` and I guess my initial thought was to return `YES` just to be sure.

On an unrelated note, NSFileManager is thread safe unless you "use a delegate to receive notifications about the status of move, copy, remove, and link operations". In which case using a shared `defaultManager` is not so safe. But I don't think that's being done here. :) https://developer.apple.com/documentation/foundation/nsfilemanager